triggers/astarte_trigger.go

```diff
@@ -0,0 +1,384 @@
+		return err
+	}
+
+	*o = AstarteHTTPMethod(j)
+	if err := o.IsValid(); err != nil {
+		return fmt.Errorf("'%v' is not a valid AstarteHTTPMethod", j)
+	}
+	return nil
+}
+
+type AstarteTriggerAction struct {
+	HTTPUrl         string            `json:"http_url"`
+	HTTPMethod      AstarteHTTPMethod `json:"http_method"`
+	HTTPHeaders     string            `json:"http_static_headers"`
+	IgnoreSslErrors bool              `default:"false"`
+}
+type AstarteSimpleTrigger struct {
+	Type               AstarteTriggerType          `json:"type"`
+	On                 AstarteTriggerOn            `json:"on"`
+	DeviceID           string                      `json:"device_id,omitempty"`
+	GroupName          string                      `json:"group_name,omitempty"`
+	InterfaceName      string                      `json:"interface_name,omitempty"`
+	InterfaceMajor     json.Number                 `json:"interface_major,omitempty"`
+	MatchPath          string                      `json:"match_path,omitempty"`
+	ValueMatchOperator AstarteTriggerMatchOperator `json:"value_match_operator"`
+	KnownValue         *json.Number                `json:"known_value,omitempty"`
+}
+
+// AstarteInterface represents an Astarte Interface
+type AstarteTrigger struct {
+	Name           string                 `json:"name"`
+	Action         AstarteTriggerAction   `json:"action"`
+	SimpleTriggers []AstarteSimpleTrigger `json:"simple_triggers"`
+}
+
+// requiredAstarteTrigger is an helper struct used for validating required fields when unmarshalling an
+// astarte interface. Its fields are defined as pointers so that it is possible determining if any field is
+// present and valid.
+type requiredAstarteTrigger struct {
+	Name           *string                        `json:"name"`
+	Action         *requiredAstarteTriggerAction  `json:"action"`
+	SimpleTriggers []requiredAstarteSimpleTrigger `json:"simple_triggers"`
+}
+type requiredAstarteTriggerAction struct {
+	HTTPUrl    *string            `json:"http_url"`
+	HTTPMethod *AstarteHTTPMethod `json:"http_method"`
+}
+
+type requiredAstarteSimpleTrigger struct {
+	Type      *AstarteTriggerType `json:"type"`
+	On        *AstarteTriggerOn   `json:"on"`
+	DeviceID  *string             `json:"device_id,omitempty"`
+	GroupName *string             `json:"group_name,omitempty"`
+
+	InterfaceName      *string                      `json:"interface_name,omitempty"`
+	InterfaceMajor     *json.Number                 `json:"interface_major,omitempty"`
+	MatchPath          *string                      `json:"match_path,omitempty"`
+	ValueMatchOperator *AstarteTriggerMatchOperator `json:"value_match_operator"`
+	KnownValue         *json.Number                 `json:"known_value,omitempty"`
+}
+
+// ensureRequiredFields ensures that any required fields within an AstarteInterface is present and valid. It is
+// employed in place of the UnmarshalJSON interface to avoid infinite loops when unmarshalling an AstarteInterface
```

Comment:
Leftover
```suggestion
// ensureRequiredFields ensures that any required fields within an AstarteTrigger is present and valid. It is
// employed in place of the UnmarshalJSON interface to avoid infinite loops when unmarshalling an AstarteTrigger
```